crates/ruff_linter/resources/test/fixtures/flake8_type_checking/TCH007.py

```diff
@@ -0,0 +1,27 @@
+from __future__ import annotations
+
+from typing import Dict, TypeAlias, TYPE_CHECKING
+
+if TYPE_CHECKING:
+    from typing import Dict
+
+    from foo import Foo
+
+    OptStr: TypeAlias = str | None
+    Bar: TypeAlias = Foo[int]
+
+a: TypeAlias = int  # OK
+b: TypeAlias = Dict  # OK
+c: TypeAlias = Foo   # TCH007
+d: TypeAlias = Foo | None  # TCH007
+e: TypeAlias = OptStr   # TCH007
+f: TypeAlias = Bar   # TCH007
+g: TypeAlias = Foo | Bar  # TCH007 x2
+h: TypeAlias = Foo[str]  # TCH007
+
```

Comment:
What does the fix look like when comments are present?

example:
```python-console
>>> i: TypeAlias = (int |  # TCH007 x2 
float)
>>> i
int | float
```

Reply:
Until I know why this fix was marked unsafe in the other rules I will continue always to mark this fix as unsafe. If the only reason was the fix potentially eating comments I can apply the same comment detection logic to TCH007.